qcodes/instrument/base.py

```diff
@@ -74,6 +76,45 @@ def __init__(self, name: str, metadata: Optional[Mapping[Any, Any]] = None) -> N
 
         self.log = get_instrument_logger(self, __name__)
 
+        if self._call_post_init:
+            self.__post_init__(name, metadata)
+
+    def __init_subclass__(cls, **kwargs: Any) -> None:
+
+        original_init = cls.__init__
+
+        def __new_init__(
+            self: InstrumentBase, *args: Any, **sub_class_kwargs: Any
+        ) -> None:
+
+            # Call post init if `self._call_post_init`
+            # is True
+            call_post_init = self._call_post_init
+            # but prevent the init methods of
+            # base classes from also calling post init
+            self._call_post_init = False
+            original_init(self, *args, **sub_class_kwargs)
+
+            if call_post_init:
+                self.__post_init__(*args, **sub_class_kwargs)
+
+        cls.__init__ = __new_init__  # type: ignore
```

Comment:
Please use ```mypy --show-error-code``` to figure out which error code this is so that we can only skip one error code. 

`# type: ignore[nameoferrorcode] `

Reply:
I have done so now